### ISSUE TITLE
Don't dialog with device when OS locked

### ISSUE THREAD

**Issue by dr4Ke** [nthdimtech/signet-client#84]
On Windows (didn't test on other OS) the direct access slots are unusable if the signet client is executed, even if the screen is locked.
The user should be able to use these slots even in this case.

I presume that it's because of the desktop client responding to the device when running, and not a device issue.

**Comment by nthdimtech** [comment]
Good catch, I hadn't considered this. The problem is that the device is 
still in a connected state until the client is closed or the user logs 
out. I have been meaning to add an auto-logout on screen lock for awhile 
and this would solve the problem.

**Comment by nthdimtech** [comment]
This is going to be tricky. I could only find a Qt API that detects shutdowns or full logouts when the application would exit anyway. I'll have to figure this out on an OS by OS basis.

**Comment by dragon788** [comment]
@nthdimtech would it make sense to only communicate with the device during the initial application open and unlock and then during a save/refresh and keep a cached state in the app until those actions are performed? This might improve performance a little bit if you don't need to read/write all the time, but could definitely be a lot of work to implement.

**Comment by greenhelper** [comment]
I can confirm that the described behavior also happens with Ubuntu 16.04 (i.e locked screen, direct access slot not usable)

Maybe it would be possible to disconnect from the software client by pressing 5 times?

**Comment by dr4Ke** [comment]
What about the following behavior?

### Fall back to the direct access mode after a long press
- the application is running in the background (session locked or not)
- the device must be locked (user perhaps plugs it out an in again to be sure)
- long press the device (it has no meaning right now, in this state)
- the application shows a popup informing the switch to the direct access mode (or a message when using a CLI instead of a GUI)
- use the slot access mode
- switch back to the client-device communication mode by pressing a button on the interface, or restarting the application.

It should be easier to code, as it doesn't depend on the graphical environment.
